ecmascript/transforms/compat/src/es2017/async_to_generator.rs

```diff
@@ -362,6 +362,36 @@ impl Fold for Actual {
         }
     }
 
+    fn fold_module_item(&mut self, item: ModuleItem) -> ModuleItem {
+        let item = match item {
+            // if fn is ExportDefaultDecl, fn is not FnDecl but FnExpr
+            ModuleItem::ModuleDecl(ModuleDecl::ExportDefaultDecl(export_default)) => {
+                if let DefaultDecl::Fn(fn_expr) = export_default.decl {
+                    let expr = fn_expr.fold_children_with(self);
+                    let expr = if !expr.function.is_async {
+                        expr
+                    } else {
+                        let function = self.fold_fn(expr.ident.clone(), expr.function, true);
+                        FnExpr {
+                            ident: expr.ident,
+                            function,
+                        }
+                    };
+
+                    ModuleItem::ModuleDecl(ModuleDecl::ExportDefaultDecl(ExportDefaultDecl {
+                        decl: expr.into(),
+                        ..export_default
+                    }))
+                } else {
+                    ModuleItem::ModuleDecl(ModuleDecl::ExportDefaultDecl(export_default))
+                }
+            }
+            _ => item,
+        };
+
+        item.fold_children_with(self)
```

Comment:
This will fold the same node multiple times.

Reply:
I may miss something, but this seems required? otherwise moduleitem skips folds ends up not transpiling bunch of async / await with test failures.

Reply:
I mean you can add `return` to L380 to avoid duplicated folding.

Reply:
Oh :facepalm: yes, that seems do the trick.